### PR TITLE
[13.][IMP] project_timeline: add Start Date field, to replace Assigned Date (major change)

### LINKED ISSUE #774: [13.0] field "date_start" doesn't exist anymore in project_timeline
Field "date_start" dosn't exist anymore. This field appears to have been replaced by "date_assing" but this one is not editable. 
When a user want schedule a task to start on a specific date , they cannot.  The "date_assign" field is automatically set with the date corresponding to the moment in which the task is assigned to a user

### PR DESCRIPTION
**Pull request by dreispt**
Closes #774

### CODE REVIEW

File: project_timeline/hooks.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Open Source Integrators - Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+def populate_date_start(cr, registry):
```
Comment (Tardo) [inline project_timeline/hooks.py:5]: This should be moved to "migrations"

File: project_timeline/models/project_task.py [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import fields, models
 
 
 class ProjectTask(models.Model):
     _inherit = "project.task"
 
+    date_start = fields.Datetime("Start Date")
```
Comment (Tardo) [inline project_timeline/models/project_task.py:12]: To ensure avoid problems with possible function signature changes.
```suggestion
    date_start = fields.Datetime(string="Start Date")
```

**Review by Tardo — CHANGES_REQUESTED** [review]
Thanks! I think that what is being done in the hook should be moved to the migrations folder. Please, check travis errors.

**Review by anajuaristi — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@dreispt can you attend comments and make Travis green?

File: project_timeline/hooks.py (hunk 1 above)
Comment (dreispt) [reply project_timeline/hooks.py:5]: What would the cons of having it as it is?

File: project_timeline/models/project_task.py (hunk 2 above)
Comment (dreispt) [reply project_timeline/models/project_task.py:12]: I'm sorry, I don't see the point in the change here. Using positional arguments is documented and supported, and will not change in this Odoo version.

**Review by oihane — APPROVED** [review]
LGTM

File: project_timeline/hooks.py (hunk 1 above)
Comment (oihane) [reply project_timeline/hooks.py:5]: What happens if you already have it installed, there is no migration script that will execute this.

File: project_timeline/hooks.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    """
+    The date_start was introduced to be used instead of date_assign.
+    To keep same behaviour on upgrade, initialize it
+    to have the same data as before.
+    """
+    cr.execute(
+        "UPDATE project_task "
+        "SET date_start = date_assign "
+        "WHERE date_start IS NULL "
+        "AND date_assign IS NOT NULL"
```
Comment (pedrobaeza) [inline project_timeline/hooks.py:15]:
```suggestion
        "UPDATE project_task "
        "SET date_start = COALESCE(date_assign, create_date) "
        "WHERE date_start IS NULL "
```

File: project_timeline/models/project_task.py (hunk 2 above)
Comment (pedrobaeza) [reply project_timeline/models/project_task.py:12]: Not really needed, but it's always welcome to use keyword arguments as such.

File: project_timeline/hooks.py (hunk 1 above)
Comment (pedrobaeza) [reply project_timeline/hooks.py:5]: Exactly. It should be as hook, but also call it from a migration script:

```python
from odoo.addons.project_timeline.hooks import populate_date_start
from odoo.modules import registry


def migrate(cr, version):
    populate_date_start(cr, registry)
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Already done in #860

**Pull request closed without merging**